File: 프로그래머스/2/12981. 영어 끝말잇기/영어 끝말잇기.js
function solution(n, words) {
    let round = 1;
    let end = words[0].at(-1);
    let player = 1;
    const set = new Set([words[0]])
    
    for(let i=1; i<words.length; i+=1){
        player = (i%n)+1;
        round = Math.ceil((i+1)/n)
        const start = words[i][0];
        if(start!==end){
            return [player, round]
        }
        set.add(words[i]);
        if(set.size!==i+1){
            return [player, round]
        }
        end = words[i].at(-1);
    }
    
    return [0,0]
}

File: 프로그래머스/2/12981. 영어 끝말잇기/영어 끝말잇기.py
def solution(n, words):
    round = 1
    pre = None
    dup = set()
    index = 0
    for i,w in enumerate(words):
        index = i+1
        if i==0:
            pre = w
            dup.add(w)
            continue
        
        if i%n==0: round+=1
        
        if w[0] != pre[-1]:
            break
        
        dup_size = len(dup)
        dup.add(w)
        if dup_size == len(dup):
            break
        pre = w
    
    if len(dup)==len(words):
        return [0,0]
    
    return [index-(round-1)*n, round]